hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/DagBuildContextImpl.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2023 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl.opt.physical;
+
+import com.hazelcast.jet.core.DAG;
+import com.hazelcast.jet.impl.util.Util;
+import com.hazelcast.jet.sql.impl.connector.SqlConnector.DagBuildContext;
+import com.hazelcast.jet.sql.impl.opt.OptUtils;
+import com.hazelcast.sql.impl.QueryParameterMetadata;
+import com.hazelcast.sql.impl.expression.Expression;
+import com.hazelcast.sql.impl.plan.node.PlanNodeFieldTypeProvider;
+import com.hazelcast.sql.impl.schema.Table;
+import org.apache.calcite.rex.RexNode;
+import org.apache.calcite.rex.RexVisitor;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+
+public class DagBuildContextImpl implements DagBuildContext {
+    private final DAG dag;
+    private final QueryParameterMetadata parameterMetadata;
+    private Table table;
+    private PhysicalRel rel;
+
+    public DagBuildContextImpl(DAG dag, QueryParameterMetadata parameterMetadata) {
+        this.dag = dag;
+        this.parameterMetadata = parameterMetadata;
+    }
+
+    @Nonnull
+    @Override
+    public DAG getDag() {
+        return dag;
+    }
+
+    @Nullable
+    @Override
+    public Table getTable() {
+        return table;
```

Comment:
maybe throw exception/assert if someone want to get table and it is null?

Reply:
Yeah, currently we can do this. In some contexts it's always available and in other never, so the user code doesn't need to test for its presence.